Mongo read preference

SamBartrum opened the pull request:
A potential fix relating to this issue: https://github.com/compose/transporter/issues/392

Expose the ability to set the mongo read preference within the connection config.

(This is my first time writing any go, so go easy on me!)

### Required for all PRs:

- [ ] CHANGELOG.md updated (feel free to wait until changes have been reviewed by a maintainer)
- [x] README.md updated (if needed)

jipperinbham commented:
this looks great and inline with the general code style

I took a cursory look at the failing tests and will try to reproduce locally.

SamBartrum commented:
Excellent. I'll try and figure out what is going wrong with the tests too - although not had much luck so far.

jipperinbham commented on adaptor/mongodb/client.go:
```diff
@@ -189,6 +194,30 @@ func WithTail(tail bool) ClientOptionFunc {
+			c.readPreference = DefaultReadPreference
+			return nil
+		}
+		switch read_preference {
```
I'm thinking it'd be nice to do a `strings.ToLower(read_preference)` here so the preference is insensitive.

jipperinbham commented on adaptor/mongodb/client_test.go:
```diff
@@ -218,9 +230,36 @@ var clientTests = []struct {
+			readPreference: 2,
+		},
+		nil,
+	},
```
please add test cases to cover all possible preferences

jipperinbham commented on client/error.go:
```diff
@@ -49,3 +49,12 @@ func (e VersionError) Error() string {
 	}
 	return fmt.Sprintf("%s running %s, %s", e.URI, e.V, e.Err)
 }
+
```
this should be in `adaptor/mongodb/client.go` (below [here](https://github.com/compose/transporter/blob/master/adaptor/mongodb/client.go#L37-L44)) since it's specific to mongodb

jipperinbham commented:
LGTM, if you could rebase from master and preferably squash the commits, I'll get this merged in

SamBartrum commented:
Just added a last minute test for the InvalidReadPreferenceError and fixed a typo in a test. It should  now be rebased from master and squashed into a single commit. Good to go?
